widgetssdk/src/main/java/com/glia/widgets/chat/Intention.kt

```diff
@@ -0,0 +1,37 @@
+package com.glia.widgets.chat
+
+internal enum class Intention {
+    /* Restore Chat. Used in cases when navigating from chat screen directly, or from other screens with bubble */
+    RESTORE_CHAT,
+
+    /* Open Secure Conversation chat with Leave dialog, and if the visitor chooses to leave, open chat screen and start enqueueing for audio */
+    SC_DIALOG_START_AUDIO,
+
+    /* Open Secure Conversation chat with Leave dialog, and if the visitor chooses to leave, open chat screen and  start enqueueing for video */
+    SC_DIALOG_START_VIDEO,
+
+    /* Open Secure Conversation chat with Leave dialog, and if the visitor chooses to leave, start enqueueing for chat */
+    SC_DIALOG_ENQUEUE,
```

Comment:
@DavDo Not critical. Will it be better to rename `SC_DIALOG_ENQUEUE` to `SC_DIALOG_START_CHAT` to align with similar flows `SC_DIALOG_START_AUDIO` and `SC_DIALOG_START_VIDEO`?

Reply:
Good point!
I also came to that conclusion and renamed it in my ongoing task to `SC_DIALOG_START_ENQUEUEING`, this will be a bit clearer since we will be in the chat screen, but when the visitor chooses to leave, we must start enqueueing.

Reply:
Yes. The matter is that ENQUEUE might mean any type of engagement: enqueue to chat/audio/video, isn't it? That's why I prefer `SC_DIALOG_START_CHAT` or `SC_DIALOG_ENQUEUE_CHAT`. And I would suggest to align the names for all these cases: for chat, audio, video.

Reply:
But from the chat screen point, these cases are different.

- SC_DIALOG_START_AUDIO, SC_DIALOG_START_VIDEO → will open the call screen
- SC_DIALOG_START_ENQUEUEING → will start enqueueing

Having this said, I'd prefer `SC_DIALOG_ENQUEUE_FOR_TEXT` since the engagement type is `Engagement.MediaType.TEXT`

Reply:
Okay, I think I got your point now. Agree.